app/src/main/java/io/gnosis/safe/notifications/NotificationRepository.kt

```diff
@@ -72,125 +65,102 @@ class NotificationRepository(
         notificationManager.hideAll()
     }
 
-    suspend fun register() {
-        if (deviceUuid == null || registrationUpdateFailed) {
-            kotlin.runCatching {
-                getCloudMessagingToken()
-            }
-                .onSuccess {
-                    it?.let {
-                        register(it)
-                    }
-                }
-        }
-    }
-
-    suspend fun register(token: String) {
+    suspend fun register(token: String? = null) {
+        val cloudMessagingToken = token ?: getCloudMessagingToken()
+        cloudMessagingToken?.let {
+            val registration = Registration(
+                uuid = deviceUuid ?: generateUUID(),
+                cloudMessagingToken = cloudMessagingToken,
+                buildNumber = BuildConfig.VERSION_CODE.toString(),
+                bundle = BuildConfig.APPLICATION_ID,
+                deviceType = "ANDROID",
+                version = appVersion,
+                timestamp = (System.currentTimeMillis() / 1000).toString()
+            )
 
-        val safes = getSafeStrings()
+            val safesByChain = safeRepository.getSafes().groupBy { it.chainId }
 
-        val registration = Registration(
-            uuid = deviceUuid ?: generateUUID(),
-            safes = safes,
-            cloudMessagingToken = token,
-            bundle = BuildConfig.APPLICATION_ID,
-            deviceType = "ANDROID",
-            version = appVersion,
-            buildNumber = BuildConfig.VERSION_CODE.toString(),
-            timestamp = (System.currentTimeMillis() / 1000).toString()
-        )
+            if (safesByChain.isNotEmpty()) {
 
-        registration.addSignatures(credentialsRepository.owners())
+                val owners = credentialsRepository.owners()
+                registration.addChainRegistrations(safesByChain, owners)
 
-        if (safes.isNotEmpty()) {
-            kotlin.runCatching {
-                notificationService.register(registration)
-            }
-                .onSuccess {
-                    Timber.d("notification service registration success")
-                    deviceUuid = it.uuid
-                    registrationUpdateFailed = false
-                }
-                .onFailure {
-                    Timber.d("notification service registration failure")
-                    deviceUuid = null
-                    handleCloudMessagingTokenIsLinkedToAnotherDeviceError(it)
-                    registrationUpdateFailed = true
+                kotlin.runCatching {
+                    notificationService.register(registration)
                 }
+                    .onSuccess {
+                        Timber.d("notification service registration success")
+                        deviceUuid = registration.uuid
```

Comment:
didn't we say that notification service won't return the uuid anymore and we would just generate it in the app and submit it?

Reply:
it's not returned. Here locally generated uuid is saved if request is successful